libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -0,0 +1,293 @@
+          ? '#FC8B8B'
+          : number > 10
+          ? '#FDC4C4'
+          : number > 1
+          ? '#FDE6E6'
+          : '#FFFFFF';
+      stateExpression.push(row['STATE'], color);
+    });
+
+    this.countyData[this.selectedDate].forEach(function(row) {
+      const number = row['infection_rate'];
+      const color =
+        number > 1000
+          ? '#AE8080'
+          : number > 500
+          ? '#D18080'
+          : number > 100
+          ? '#FC8B8B'
+          : number > 10
+          ? '#FDC4C4'
+          : number > 1
+          ? '#FDE6E6'
+          : '#FFFFFF';
+      countyExpression.push(row['fips'], color);
+    });
+
+    stateExpression.push('rgba(255,255,255,1)');
+    countyExpression.push('rgba(255,255,255,1)');
+
+    this.mapService.loaded.subscribe((map) => {
+      map.setPaintProperty('covid-state', 'fill-color', stateExpression);
+      map.setPaintProperty('covid-county', 'fill-color', countyExpression);
+    });
+
+    //this.mapService.map.setPaintProperty('covid-state', 'fill-color', stateExpression);
+    //this.mapService.map.setPaintProperty('covid-county', 'fill-color', countyExpression);
+  }
+
+  public drawDeathMap() {
+    console.log('death');
+
+    const stateExpression = ['match', ['get', 'STATE']];
+    const countyExpression = ['match', ['get', 'fips']];
+
+    this.stateData[this.selectedDate].forEach(function(row) {
+      const number = row['death_rate'];
+      const color =
+        number > 100
+          ? '#54278f'
+          : number > 75
+          ? '#756bb1'
+          : number > 50
+          ? '#9e9ac8'
+          : number > 10
+          ? '#bcbddc'
+          : number > 1
+          ? '#dadaeb'
+          : '#f2f0f7';
+      stateExpression.push(row['STATE'], color);
+    });
+
+    this.countyData[this.selectedDate].forEach(function(row) {
+      const number = row['death_rate'];
+      const color =
+        number > 100
+          ? '#54278f'
+          : number > 75
+          ? '#756bb1'
+          : number > 50
+          ? '#9e9ac8'
+          : number > 10
+          ? '#bcbddc'
+          : number > 1
+          ? '#dadaeb'
+          : '#f2f0f7';
+      countyExpression.push(row['fips'], color);
+    });
+
+    stateExpression.push('rgba(255,255,255,1)');
+    countyExpression.push('rgba(255,255,255,1)');
+
+    this.mapService.loaded.subscribe((map) => {
+      map.setPaintProperty('covid-state', 'fill-color', stateExpression);
+      map.setPaintProperty('covid-county', 'fill-color', countyExpression);
+    });
+  }
+
+  public stateToggle(toggleButton) {
+    this.stateButtonToggle = !this.stateButtonToggle;
+    this.switchLayers();
+  }
+
+  public mortalityToggle(toggleButton) {
+    this.mortalButtonToggled = !this.mortalButtonToggled;
+    this.mortalButtonToggled ? this.drawDeathMap() : this.drawCasesMap();
+    if(this.mortalButtonToggled){
+      this.drawDeathMap()
+      document.getElementById('cases-legend').style.display = 'none';
```

Comment:
Same thing here with using `document`.

Additionally, avoid setting styles in code. Use CSS classes or template directives like `NgClass`, `NgStyle`, or other structural directives such as `NgIf`.

Since it seems all you want to do is hide/show something, use `NgIf`

```html
    <div *ngIf="mortalButtonToggled ">
         <!-- The ngIf directive will evaluate the mortalButtonToggled value and hide/show based on its coerced boolean value. This has the benefit of completely removing the element from the DOM when the expression evaluates to false. -->
    </div>
```

Reply:
used ngif on the legend element